lesson_07/homework_07.py

```diff
@@ -0,0 +1,173 @@
+print("\nTask01.\nЗадача - надрукувати табличку множення на задане число, але лише до максимального значення "
+      "для добутку - 25. Код майже готовий, треба знайти помилки та випраавити/доповнити.")
+
+
+def multiplication_table(number):
+    # Initialize the appropriate variable
+    multiplier = 1
+    max_value = 25
+    result = 0
+
+    # Complete the while loop condition.
+    while result < max_value:
+        result = number * multiplier
+        # десь тут помила, а може не одна
+        if result > max_value:
+            # Enter the action to take if the result is greater than 25
+            break
+        print(str(number) + "x" + str(multiplier) + "=" + str(result))
+
+        # Increment the appropriate variable
+        multiplier += 1
+
+
+multiplication_table(3)
+# Should print:
+# 3x1=3
+# 3x2=6
+# 3x3=9
+# 3x4=12
+# 3x5=15
+
+
+print("\nTask02.\nНаписати функцію, яка обчислює суму двох чисел.")
+
+
+def sum_of_2_nums(num_1: int, num_2: int) -> int:
+    return num_1 + num_2
+
+
+print(sum_of_2_nums(2, 3))
+
+
+print("\nTask03.\nНаписати функцію, яка розрахує середнє арифметичне списку чисел.")
+
+
+def middle_of_nums(nums_list: list) -> float:
+    return sum(nums_list)/len(nums_list)
+
+
+print(middle_of_nums([1, 2, 3, 4, 5]))
+
+
+print("\nTask04.\nНаписати функцію, яка приймає рядок та повертає його у зворотному порядку.")
+
+
+def revers_list(new_string: str) -> str:
+    return new_string[::-1]
```

Comment:
On the last lection we have learned method for reversing any iterable element. It is more clear than slicing.

Reply:
Updated code using the building function "reversed" in the second commit.